sample-apps/solid_queue-sample/db/schema.rb

```diff
@@ -0,0 +1,136 @@
+# This file is auto-generated from the current state of the database. Instead
+# of editing this file, please use the migrations feature of Active Record to
+# incrementally modify your database, and then regenerate this schema definition.
+#
+# This file is the source Rails uses to define your schema when running `bin/rails
+# db:schema:load`. When creating a new database, `bin/rails db:schema:load` tends to
+# be faster and is potentially less error prone than running all of your
+# migrations from scratch. Old migrations may fail to apply correctly if those
+# migrations use external dependencies or application code.
+#
+# It's strongly recommended that you check this file into your version control system.
+
+ActiveRecord::Schema[7.1].define(version: 2024_04_23_195527) do
+  create_schema "_timescaledb_cache"
+  create_schema "_timescaledb_catalog"
+  create_schema "_timescaledb_config"
+  create_schema "_timescaledb_internal"
+  create_schema "timescaledb_experimental"
+  create_schema "timescaledb_information"
+  create_schema "toolkit_experimental"
```

Comment:
These were causing me trouble locally when running `db:prepare` (via `bin/setup`):

```sh
solid_queue-sample $ bin/rails db:prepare
   (1.5ms)  CREATE SCHEMA "_timescaledb_cache"
bin/rails aborted!
ActiveRecord::StatementInvalid: PG::DuplicateSchema: ERROR:  schema "_timescaledb_cache" already exists (ActiveRecord::StatementInvalid)
/Users/adam/Projects/judoscale-ruby/sample-apps/solid_queue-sample/db/schema.rb:14:in `block in <top (required)>'
/Users/adam/Projects/judoscale-ruby/sample-apps/solid_queue-sample/db/schema.rb:13:in `<top (required)>'

Caused by:
PG::DuplicateSchema: ERROR:  schema "_timescaledb_cache" already exists (PG::DuplicateSchema)
/Users/adam/Projects/judoscale-ruby/sample-apps/solid_queue-sample/db/schema.rb:14:in `block in <top (required)>'
/Users/adam/Projects/judoscale-ruby/sample-apps/solid_queue-sample/db/schema.rb:13:in `<top (required)>'
Tasks: TOP => db:prepare
(See full trace by running task with --trace)
```

I deleted these lines and ran `db:prepare` successfully, but the lines were automatically added back to `schema.rb`.

This is only a problem on first-time setup, but it's annoying. It's also Timescale-specific, which our sample apps don't require at all.

I'm not really sure any way around it, though. If you try to set up a sample app while connected to Postgres with Timescale enabled, you'll get these lines in your schema. 🤔

Reply:
I think I saw some timescale stuff dumped to the schema, however some output was also in the good_job sample app and I didn't pay much attention, but now I can see that `create_schema` is only on this one... turns out it's something that was added to Rails 7.1:
- https://github.com/rails/rails/pull/49164

So when recreating the DB, it will try to recreate these schemas and fail... it works with `enable_extension` because I believe that adds a "IF NOT EXISTS", but `create_schema` apparently does not do anything like that... maybe it should.

There's some potentially related changes: 
- https://github.com/rails/rails/pull/49555
- https://github.com/rails/rails/pull/48132

Other than monkey-patching Rails / schema dumper, I can't really think of an option right now 🤔

Reply:
Thanks for the digging! Let's just leave it for now. I think we're the only ones using these sample apps.